### ISSUE TITLE
Kics not detecting USER instruction in a multi-stage Dockerfile

### ISSUE THREAD

**Issue by opalmer** [Checkmarx/kics#4850]
Hi,

Our builds typically follow a multi-stage build process. Below is an example Dockerfile:

```Dockerfile
FROM alpine:latest as builder

# Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.
RUN touch /test

FROM alpine:latest
COPY --from=builder /test /test
RUN addgroup app && adduser -s /bin/false -h /doesnotexist -D -H -G app app

USER app
```

When kics scans the above we get:

```
Missing User Instruction, Severity: HIGH, Results: 1
Description: A user should be specified in the dockerfile, otherwise the image will run as root
Platform: Dockerfile

	[1]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.
```

### Expected Behavior

kics doesn't raise a "Missing User Instruction" finding.

### Actual Behavior

(Formatted logs and samples helps us to better understand the issue)

```
$ cat Dockerfile
FROM alpine:latest as builder

# Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.
RUN touch /test

FROM alpine:latest
COPY --from=builder /test /test
RUN addgroup app && adduser -s /bin/false -h /doesnotexist -D -H -G app app

USER app

$ kics scan -p ./Dockerfile -t dockerfile -o ./results -- report-formats json -v

                   .0MO.
                   OMMMx
                   ;NMX;
                    ...           ...              ....
WMMMd     cWMMM0.  KMMMO      ;xKWMMMMNOc.     ,xXMMMMMWXkc.
WMMMd   .0MMMN:    KMMMO    :XMMMMMMMMMMMWl   xMMMMMWMMMMMMl
WMMMd  lWMMMO.     KMMMO   xMMMMKc...'lXMk   ,MMMMx   .;dXx
WMMMd.0MMMX;       KMMMO  cMMMMd        '    'MMMMNl'
WMMMNWMMMMl        KMMMO  0MMMN               oMMMMMMMXkl.
WMMMMMMMMMMo       KMMMO  0MMMX                .ckKWMMMMMM0.
WMMMMWokMMMMk      KMMMO  oMMMMc              .     .:OMMMM0
WMMMK.  dMMMM0.    KMMMO   KMMMMx'    ,kNc   :WOc.    .NMMMX
WMMMd    cWMMMX.   KMMMO    kMMMMMWXNMMMMMd .WMMMMWKO0NMMMMl
WMMMd     ,NMMMN,  KMMMO     'xNMMMMMMMNx,   .l0WMMMMMMMWk,
xkkk:      ,kkkkx  okkkl        ;xKXKx;          ;dOKKkc


Scanning with Keeping Infrastructure as Code Secure 1.5.1


Preparing Scan Assets: Done
Executing queries: [---------------------------------------------------] 100.00%

Files scanned: 1
Parsed files: 1
Queries loaded: 52
Queries failed to execute: 0

------------------------------------

Healthcheck Instruction Missing, Severity: LOW, Results: 2
Description: Ensure that HEALTHCHECK is being used. The HEALTHCHECK instruction tells Docker how to test a container to check that it is still working
Platform: Dockerfile

	[1]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.


	[2]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.


Image Version Using 'latest', Severity: MEDIUM, Results: 2
Description: When building images, always tag them with useful tags which codify version information, intended destination (prod or test, for instance), stability, or other information that is useful when deploying the application in different environments. Do not rely on the automatically-created latest tag
Platform: Dockerfile

	[1]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.


	[2]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.


Missing User Instruction, Severity: HIGH, Results: 1
Description: A user should be specified in the dockerfile, otherwise the image will run as root
Platform: Dockerfile

	[1]: ./Dockerfile:1

		001: FROM alpine:latest as builder
		002:
		003: # Usually this is a `go build` or other instruction that produces a binary we can copy to the final layer.



Results Summary:
HIGH: 1
MEDIUM: 2
LOW: 2
INFO: 0
TOTAL: 5

Results saved to file results/results.json
Generating Reports: Done
Scan duration: 1.664091864s
A new version 'v1.5.2' of KICS is available, please consider updating
```

### Steps to Reproduce the Problem

(Command line arguments and flags used)

1.  Copy the above example file to `Dockerfile`
2.  Clone down the kics repo, checkout the 1.5.1 tag, build the binary and symlink github.com/Checkmarx/kics/assets/ to ./assets
3.  Run `kics scan -p ./Dockerfile -t dockerfile -o ./results -- report-formats json -v`

Report produced: [results.tar.gz](https://github.com/Checkmarx/kics/files/8082352/results.tar.gz)

### Specifications
(N/A if not applicable)

-   Version: 1.5.1
-   Platform: OS X but this can reproduced on other platforms.
-   Subsystem: N/A

**Comment by nunoocx** [comment]
Hi @opalmer, thanks for raising the issue. 
We are aware of it and will work on this really soon, improving all our KICS Docker queries to be multi-staged aware.

**Comment by opalmer** [comment]
Awesome, that's great to hear thanks!

**Comment by rafaela-soares** [comment]
Hello, @opalmer 🙂 

We are happy to inform you that we are working on this in PR #4877!

**Comment by opalmer** [comment]
Looking forward to getting this fixed! :)

Thanks!

**Comment by rafaela-soares** [comment]
Hi, @opalmer!

The latest release (v1.5.3) has the fix 😊
